vpr/src/route/route_timing.cpp

```diff
@@ -1868,6 +2006,37 @@ void enable_router_debug(
 #endif
 }
 
+bool is_iteration_complete(bool routing_is_feasible, const t_router_opts& router_opts, int itry, std::shared_ptr<const SetupHoldTimingInfo> timing_info, bool rcv_finished) {
+    //This function checks if a routing iteration has completed.
+    //When VPR is run normally, we check if routing_budgets_algorithm is disabled, and if the routing is legal
+    //With the introduction of yoyo budgeting algorithm, we must check if there are no hold violations
+    //in addition to routing being legal and the correct budgeting algorithm being set.
+
+    if (routing_is_feasible) {
+        if (router_opts.routing_budgets_algorithm != YOYO) {
+            return true;
+        } else if (router_opts.routing_budgets_algorithm == YOYO && (timing_info->hold_worst_negative_slack() == 0 || rcv_finished) && itry != 1) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool should_setup_lower_bound_connection_delays(int itry, const t_router_opts& router_opts) {
+    //This function checks the iteration number to see if is neccessary to setup the lower bound connection delays for future comparison.
```

Comment:
Does this need a TODO?

Reply:
Not a TODO but it is a redundant function. I found that redoing route budgeting is a huge performance hit, and hasn't improved the quality of results in my experience. I can remove this function.